files/image_config/platform/rc.local

```diff
@@ -79,42 +70,15 @@ if [ -f /host/image-$sonic_version/platform/firsttime ]; then
         firsttime_exit
     fi
 
-    if [ ! -f /etc/sonic/init_cfg.json ]; then
-        # Generate an empty init_cfg.json
-        echo '{}' > /etc/sonic/init_cfg.json
-    fi
-
     # Try to take old configuration saved during installation
-    if test_config; then
-        rm -f /host/old_config/sonic_version.yml
-        mv -f /host/old_config/* /etc/sonic/
-        if [ ! -f /etc/sonic/config_db.json ]; then
-            sonic-cfggen -H -m -j /etc/sonic/init_cfg.json --print-data > /etc/sonic/config_db.json
-        fi
+    # If taken, create a flag in /tmp/ to let updategraph service know
+    if [ -d /host/old_config ]; then
+        mv -f /host/old_config /etc/sonic/
+        touch /tmp/firstboot
     elif [ -f /host/minigraph.xml ]; then
-        mv /host/minigraph.xml /etc/sonic/
-        # Combine information in minigraph and init_cfg.json to form initiate config DB dump file.
-        # TODO: After moving all information from minigraph to DB, sample config DB dump should be provide
-        sonic-cfggen -H -m -j /etc/sonic/init_cfg.json --print-data > /etc/sonic/config_db.json
-    else
-        # Use default minigraph.xml
-        cp /usr/share/sonic/device/$platform/minigraph.xml /etc/sonic/
-        sonic-cfggen -H -m -j /etc/sonic/init_cfg.json --print-data > /etc/sonic/config_db.json
-    fi
-
-    HWSKU=`sonic-cfggen -m /etc/sonic/minigraph.xml -v DEVICE_METADATA.localhost.hwsku`
-    if [ -f /usr/share/sonic/device/$platform/$HWSKU/buffers.json.j2 ]; then
-        # generate and merge buffers configuration into config file
-        sonic-cfggen -m -t /usr/share/sonic/device/$platform/$HWSKU/buffers.json.j2 > /tmp/buffers.json
-        sonic-cfggen -j /etc/sonic/config_db.json -j /tmp/buffers.json --print-data > /tmp/config_db.json
-        mv /tmp/config_db.json /etc/sonic/config_db.json
-
-        # Only apply qos.json when buffer configuration is available.
-        if [ -f /usr/share/sonic/device/$platform/$HWSKU/qos.json ]; then
-            # merge qos configuration into init config file
-            sonic-cfggen -j /etc/sonic/config_db.json -j /usr/share/sonic/device/$platform/$HWSKU/qos.json --print-data > /tmp/config_db.json
-            mv /tmp/config_db.json /etc/sonic/config_db.json
-        fi
+        mkdir -p /etc/sonic/old_config
+        mv /host/minigraph.xml /etc/sonic/old_config/
+        touch /tmp/firstboot
```

Comment:
>firstboot [](start = 19, length = 9)

why do we need this firstboot, can updategraph check the /etc/sonic/old_config directory for it to pick up the minigraph or config_db from that directory?

Reply:
The plan is to keep `/etc/sonic/old_config` folder there and not to delete it after first boot for easy debugging. Therefore we need a flag to mark whether it is the first boot.

Reply:
I feel like the file name `firstboot` is confusing because we also reference `/host/image-$sonic_version/platform/firsttime` in this file, which is also used to determine whether it's the initial boot.

Can we give this file a more specific name based on its application?

Also, I personally think `firstboot` is a better name for `/host/image-$sonic_version/platform/firsttime`